controller/Results.php

```diff
@@ -44,64 +44,103 @@
 class Results extends tao_actions_SaSModule
 {
 
+    private $deliveryService;
+    
     /**
      * constructor: initialize the service and the default data
      * @return Results
      */
     public function __construct()
     {
         parent::__construct();
-
+        
+        $this->service = ResultsService::singleton();
+        $this->deliveryService = DeliveryAssemblyService::singleton();
         $this->defaultData();
     }
+    
+    public function getRootClass()
+    {
+        return $this->deliveryService->getRootClass();
+    }
 
     /**
      * @return ResultsService
      */
     protected function getClassService()
     {
-        return ResultsService::singleton();
+        return $this->service;
     }
 
-    /**
-     * Get all delivery execution to feed the tree
-     * @throws \common_exception_IsAjaxAction
-     */
     public function getOntologyData()
     {
         if (!tao_helpers_Request::isAjax()) {
             throw new common_exception_IsAjaxAction(__FUNCTION__);
         }
-        
-        $instances = array();
-        $deliveryService = DeliveryAssemblyService::singleton();
-        if (!$this->hasRequestParameter('classUri') || $deliveryService->getRootClass()->getUri() === $this->getRequestParameter('classUri')) {
-            // root
-            foreach ($deliveryService->getAllAssemblies() as $assembly) {
-                $child["attributes"] = array(
-                    "id" => tao_helpers_Uri::encode($assembly->getUri()),
-                    "class" => "node-class",
-                    'data-uri' => $assembly->getUri()
-                );
-                $child["data"] = htmlentities($assembly->getLabel());
-                $child["type"] = "class";
-
-                $instances[] = $child;
+
```

Comment:
You are using the same code than in `tao_actions_RdfController` when this class extends it. Any reason for that ?